pandas/tests/arrays/integer/test_dtypes.py

```diff
@@ -269,6 +269,14 @@ def test_to_numpy_dtype(dtype, in_series):
     tm.assert_numpy_array_equal(result, expected)
 
 
+@pytest.mark.parametrize("dtype", ["int32", "int64"])
+def test_nd_array_cast_to_dtype(dtype):
+    b = pd.DataFrame([["0", "1"], ["2", "3"]], dtype="Int64")
+    result = b.to_numpy(dtype=dtype)
+    expected = np.array([[0, 1], [2, 3]], dtype=dtype)
+    tm.assert_numpy_array_equal(result, expected)
```

Comment:
This does not test the actual problem, i.e. this test will pass even without your fix.
Could you modify the test to create a `DataFrame` with `int32` dtype

Reply:
Ok Sure.